src/components/Faucet/AmountField.tsx

```diff
@@ -0,0 +1,57 @@
+import React, { useMemo } from "react";
+import { StatusContext, TokenType } from "../../lib/Types";
+import { autoSelectInputText } from "../../lib/Utils";
+import { Input } from "../UI/Input/Input";
+import Config from "../../Config";
+import { tokensLabels } from "./Faucet.const";
+import { formatNumber } from "../../utils/formaters";
+import { FormState } from "./Faucet";
+
+const { minMav, maxMav } = Config.application;
+
+type Props = {
+  status: StatusContext;
+  setFormState: (
+    value: ((prevState: FormState) => FormState) | FormState,
+  ) => void;
+  formState: FormState;
+};
+
+export function AmountField(props: Props) {
+  const { status, setFormState, formState } = props;
+
+  const validateAmount = (amount: number) =>
+    amount >= minMav && amount <= maxMav;
+
+  const updateAmount = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const value = e.target.value.slice(0, 16);
+    const numberValue = Number(value);
+    if (numberValue === 0 || validateAmount(numberValue)) {
+      setFormState((prevState) => ({ ...prevState, tokenAmount: value }));
```

Comment:
better to use bignumber js when working with amount where it has some fixed decimals after comma (f.e. 6.213788)